### PR TITLE
Handle error during initial fetching inside `useThreads` and `useThreadsSuspense`

### PR DESCRIPTION
**Pull request by nimeshnayaju**
Stores any error that occurs during the initial fetch inside `useThreads` and `useThreadsSuspense`.

- [x] `useThreads` should return an error if the initial fetch is failing
- [x] `useThreadsSuspense` should throw an error if the initial fetch is failing
- [x] Integration tests

### CODE REVIEW

**Comment by nimeshnayaju** [comment]
@GuillaumeSalles `useThreads` and `useThreadsSuspense` both handle error that are encountered during the initial fetch. Any thoughts on how we could deal with errors thrown during polling and comment Websocket message handling? We've done retries using exponential backoff algorithm on error in our previous implementation.

File: packages/liveblocks-react/src/room.tsx [hunk 1]
```diff
@@ -1085,31 +1085,30 @@ export function createRoomContext<
...
+        ),
+        queryKey
+      );
 
-    poller.start(POLLING_INTERVAL);
+      poller.start(POLLING_INTERVAL);
+    } catch (err) {
+      store.setThreadsQueryState(queryKey, {
+        isLoading: false,
+        error: err as Error,
```
Comment (nimeshnayaju) [inline packages/liveblocks-react/src/room.tsx:1109]: **TODO**: If the initial request failed but polling was successful, we want to clear this `error` variable.

Comment (nimeshnayaju) [reply packages/liveblocks-react/src/room.tsx:1108]: Done: https://github.com/liveblocks/liveblocks/pull/1375/commits/7cbfa0ea5ee994ead736f709e21e84190eac875f

**Review by marcbouchenoire — APPROVED** [review]
Looking good from my perspective.

**Pull request merged**